### PR TITLE
Serverless hello user

### PR DESCRIPTION
**Pull request by Reltre**
I've created this POST endpoint for the `helloUser` function. I wasn't entirely sure based on the card's description how you wanted to pass JSON data to the `helloUser` function, so I created a `data.json` file under the `lib` directory. With that in mind, the `helloUser` function has to be invoked by specifying a path to that data file with the -p flag. I've tested this locally and in production and it all seems to work. 

Local: `serverless invoke local -p ./lib/data.json -f helloUser`  
Production: `serverless invoke -p ./lib/data.json -f helloUser`

I had to change the `event` type from `APIGatewayEvent` to `any` since `event` is being used to pass in that json data now. Let me know if you require any changes and I'll make them. Looking forward to your review!

Last thing, I was wondering, when deploying and making an update to a project, is it ok to use `-f` option? I was thinking since I was adding a function that I should deploy that way to save a bit of time, but I wasn't sure if that would deploy it to a different location, so I just used `serverless deploy` instead (in retrospect it probably would have worked out just fine). Anyway, just wanted to ask to be sure about this.

### CODE REVIEW

File: src/handlers/handler.ts [hunk 1]
```diff
@@ -11,3 +11,15 @@ export const helloWorld: Handler = (event: APIGatewayEvent, context: Context, ca
 
   callback(null, response);
 }
+
+export const helloUser: Handler = (event: any, context: Context, callback: Callback) => {
```
Comment (anonmos) [inline src/handlers/handler.ts]: Keeping the event type as `APIGatewayEvent` will actually work for our uses.  We're after `event.body.name`, which should be part of the definition for that type.

File: src/handlers/handler.ts [hunk 2]
```diff
@@ -11,3 +11,15 @@ export const helloWorld: Handler = (event: APIGatewayEvent, context: Context, ca
...
+  const response = {
+    statusCode: 200,
+    body: JSON.stringify({
+      message: `Hello ${event.name}!`,
```
Comment (anonmos) [inline src/handlers/handler.ts]: I'm actually after being able to POST the name to this endpoint using a REST client.  The results from a test post look like this:
`{
  "message": "Hello undefined!",
  "input": {
    "resource": "/hello-user",
    "path": "/hello-user",
    "httpMethod": "POST",
    "headers": {
      "Accept": "*/*",
      "Accept-Encoding": "gzip, deflate, br",
      "Accept-Language": "en-US,en;q=0.5",
      "CloudFront-Forwarded-Proto": "https",
      "CloudFront-Is-Desktop-Viewer": "true",
      "CloudFront-Is-Mobile-Viewer": "false",
      "CloudFront-Is-SmartTV-Viewer": "false",
      "CloudFront-Is-Tablet-Viewer": "false",
      "CloudFront-Viewer-Country": "US",
      "content-type": "text/plain;charset=UTF-8",
      "dnt": "1",
      "Host": "hc9pdzn4kc.execute-api.us-east-1.amazonaws.com",
      "origin": "moz-extension://913b8cf5-56c5-4a45-8e9b-87c73c41b967",
      "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.13; rv:57.0) Gecko/20100101 Firefox/57.0",
      "Via": "2.0 0576b942ae9f4fc9c0b62b0736e9bfd6.cloudfront.net (CloudFront)",
      "X-Amz-Cf-Id": "0OJoIuyglBJWL_wSp_8560yfLOYSbLrqlOjYJYHu1yMiCPucen76hg==",
      "X-Amzn-Trace-Id": "Root=1-5a6d4292-0660e3651a88d4783087d500",
      "X-Forwarded-For": "73.93.143.9, 54.239.203.107",
      "X-Forwarded-Port": "443",
      "X-Forwarded-Proto": "https"
    },
    "queryStringParameters": null,
    "pathParameters": null,
    "stageVariables": null,
    "requestContext": {
      "requestTime": "28/Jan/2018:03:25:06 +0000",
      "path": "/dev/hello-user",
      "accountId": "078302068589",
      "protocol": "HTTP/1.1",
      "resourceId": "zhlh0t",
      "stage": "dev",
      "requestTimeEpoch": 1517109906383,
      "requestId": "d60e85f0-03da-11e8-b5e8-2952b3373c49",
      "identity": {
        "cognitoIdentityPoolId": null,
        "accountId": null,
        "cognitoIdentityId": null,
        "caller": null,
        "sourceIp": "73.93.143.9",
        "accessKey": null,
        "cognitoAuthenticationType": null,
        "cognitoAuthenticationProvider": null,
        "userArn": null,
        "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.13; rv:57.0) Gecko/20100101 Firefox/57.0",
        "user": null
      },
      "resourcePath": "/hello-user",
      "httpMethod": "POST",
      "apiId": "hc9pdzn4kc"
    },
    "body": "{\"name\":\"Tim\"}",
    "isBase64Encoded": false
  }
}`

So you should be able to use the value from `event.body.name`.

**Review by anonmos — CHANGES_REQUESTED** [review]
Close!  A couple changes and it'll be ready to roll.

**Comment by anonmos** [comment]
Responding to this question:

> Last thing, I was wondering, when deploying and making an update to a project, is it ok to use -f option? I was thinking since I was adding a function that I should deploy that way to save a bit of time, but I wasn't sure if that would deploy it to a different location, so I just used serverless deploy instead (in retrospect it probably would have worked out just fine). Anyway, just wanted to ask to be sure about this.

Yep!  That should work fine, the region is defined in `serverless.yml` at `region: us-west-1`.  So it should work regardless.

**Comment by Reltre** [comment]
I've made some changes and the function does seem to work now with a REST client. I realized that I have to parse the JSON first, due to the fact that the default config for serverless uses lambda-proxy. 

I couldn't seem to test this from the CLI by passing json as a raw string. I guess when you invoke a function this way, `event` is just set to a JSON object and not to an `APIGatewayEvent`. 

Anyway, let me know what you think! I used Paw REST client to test the helloUser function.

**Review by anonmos — APPROVED** [review]
👍 LGTM!

**Pull request merged**